pkg/volume/csi/csi_mounter.go

```diff
@@ -98,33 +99,41 @@ func (c *csiMountMgr) CanMount() error {
 	return nil
 }
 
-func (c *csiMountMgr) SetUp(mounterArgs volume.MounterArgs) error {
-	return c.SetUpAt(c.GetPath(), mounterArgs)
+func (c *csiMountMgr) SetUp(mounterArgs volume.MounterArgs) (volumetypes.OperationStatus, error) {
+	opExitStatus, err := c.setupUtil(c.GetPath(), mounterArgs)
+	return opExitStatus, err
 }
 
 func (c *csiMountMgr) SetUpAt(dir string, mounterArgs volume.MounterArgs) error {
+	_, err := c.setupUtil(dir, mounterArgs)
+	return err
+}
+
+func (c *csiMountMgr) setupUtil(dir string, mounterArgs volume.MounterArgs) (volumetypes.OperationStatus, error) {
 	klog.V(4).Infof(log("Mounter.SetUpAt(%s)", dir))
+	// default to finished operation status
+	opExitStatus := volumetypes.OperationFinished
 
 	mounted, err := isDirMounted(c.plugin, dir)
 	if err != nil {
-		return errors.New(log("mounter.SetUpAt failed while checking mount status for dir [%s]: %v", dir, err))
+		return opExitStatus, errors.New(log("mounter.SetUpAt failed while checking mount status for dir [%s]: %v", dir, err))
 	}
 
 	if mounted {
 		klog.V(4).Info(log("mounter.SetUpAt skipping mount, dir already mounted [%s]", dir))
-		return nil
+		return opExitStatus, nil
 	}
 
 	csi, err := c.csiClientGetter.Get()
 	if err != nil {
-		return errors.New(log("mounter.SetUpAt failed to get CSI client: %v", err))
+		return opExitStatus, errors.New(log("mounter.SetUpAt failed to get CSI client: %v", err))
```

Comment:
is it possible that temporarily failt to get csiClient, so that status should return NoChange?

Reply:
may be yeah. I will change this code.